src/containers/UsernameTranslation.js

```diff
@@ -0,0 +1,146 @@
+				throw new Error(t("ASABCredentials|Something went wrong, failed to fetch assigned credentials"));
+			}
+			const usernamesToLS = setUsernamesToLS(response.data.data, userIdsArray, cleanupTime);
+			setUsernames(usernamesToLS);
+		} catch (e) {
+			props.app.addAlert("warning", t("ASABCredentials|Something went wrong, failed to fetch assigned credentials"));
+		}
+	}
+
+	// compares array of IDs with data in localstorage
+	const matchUserIds = (credentials_id) => {
+		let currentTime = Date.now();
+		const usernamesInLS = getUsernamesFromLS('Credentials', cleanupTime);
+		let usernamesToRender = [];
+
+		if (usernamesInLS.credentials == undefined || usernamesInLS.credentials.length === 0 || usernamesInLS.expiration <= currentTime) {
+			retrieveUserNames();
+			return;
+		}
+
+		for (let i = 0; i < credentials_id.length; i++) {
+			const indexFromLS = usernamesInLS.credentials.findIndex((itemInLS) => itemInLS.id === credentials_id[i]);
+			if (indexFromLS === -1) {
+				retrieveUserNames();
+				return;
+			}
+			usernamesToRender.push({ username: usernamesInLS.credentials[indexFromLS].username, id: usernamesInLS.credentials[indexFromLS].id });
+		}
+		setUsernames(usernamesToRender);
+	}
+
+	useEffect(() => {
+		matchUserIds(userIdsArray);
+	}, [])
+
+	return (
+		<>
+			{ usernames && usernames.length !== 0 ?
+				usernames.map((userObj) => {
+					return (
+						<div title={userObj.id}>
+							<i className="cil-user pr-1"></i>
+							<Link to={{ pathname: `/auth/credentials/${userObj.id}` }}>
+								{userObj.username}
+							</Link>
+						</div>
+					)
+				})
+				:
+				userIdsArray.map((credentials_id) => {
+					return (
+						<div>
+							<i className="cil-user pr-1"></i>
+							<Link to={{ pathname: `/auth/credentials/${credentials_id}` }}>
+								{credentials_id}
+							</Link>
+						</div>
+					)
+				})
+			}
+		</>
+	)
+}
+
+// Get usernames from localstorage
+function getUsernamesFromLS(name, cleanupTime) {
+	let ls;
+	if (global.localStorage) {
+		try {
+			ls = JSON.parse(global.localStorage.getItem(name.toString()));
+		} catch (e) {
+			/*Ignore*/
+		}
+	}
+	return ls ? ls : { credentials: [], expiration: new Date().getTime() + cleanupTime };
+}
+
+function setUsernamesToLS(data, userIdsArray, cleanupTime) {
+	if (global.localStorage) {
+		let dataInLS = getUsernamesFromLS('Credentials', cleanupTime);
+		let dataToLS = [];
+		userIdsArray.map((credential_id) => {
+			let item = {};
+			if (data[credential_id]) {
+				item = {
+					id: credential_id,
+					username: data[credential_id],
+				};
+			}
+			if (!data[credential_id]) {
+				item = {
+					id: credential_id,
+					username: credential_id
+				}
+			}
+			const indexFromLS = dataInLS.credentials.findIndex((itemInLS) => itemInLS.id === item.id);
+			if (indexFromLS === -1) {
+				dataInLS.credentials.push(item);
+			}
+			dataToLS.push(item);
+		})
+		try {
+			global.localStorage.setItem('Credentials', JSON.stringify(dataInLS));
```

Comment:
You can simplify `global.localStorage.setItem(...)` to `localStorage.setItem(...)`. 
`localStorage` is already a global browser variable.

Reply:
Also I guess that there is no need to `try/catch` localStorage since it's not supported only by browsers which were released 15-20 years ago. Those browsers cannot even start our apps I guess

![Screenshot 2021-12-22 at 18 30 58](https://user-images.githubusercontent.com/37152497/147133284-b949fd18-18ed-421a-9c61-f268cfee6aa2.png)

Reply:
^^^ Those two comments are appliable to all places where you use localStorage